### PR TITLE
[FLINK-19941][Connectors / Kafka]Support sink parallelism configuration to Kafka connector

### PR DESCRIPTION
**Pull request by zhuxiaoshang**
## What is the purpose of the change

*Support sink parallelism configuration to Kafka connector*

## Brief change log

  - *Support sink parallelism configuration to Kafka connector*

## Verifying this change

This change added tests and can be verified as follows:
  - *KafkaDynamicTableFactoryTest#testSinkWithParallelism*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (no)

### CODE REVIEW

**Comment by zhuxiaoshang** [comment]
@flinkbot run azure

**Comment by zhuxiaoshang** [comment]
@flinkbot run azure

**Comment by zhuxiaoshang** [comment]
cc @JingsongLi @shouweikun

File: docs/dev/table/connectors/kafka.md [hunk 1]
```diff
@@ -184,6 +184,18 @@ Connector Options
...
+    <tr>
+      <td><h5>sink.parallelism</h5></td>
+      <td>optional</td>
+      <td style="word-wrap: break-word;">(none)</td>
+      <td>Integer</td>
+      <td>Defines the parallelism for the Kafka sink. If not specified, the parallelism are
+      <ul>
+        <li><code>Chained</code>: Use upstream parallelism.</li>
+        <li><code>Non-Chained</code>: Use global parallelism setting.</li>
+      </ul>
```
Comment (wuchong) [inline docs/dev/table/connectors/kafka.md:196]: I think it's confusing what is "parallelism are chained.". 

What do you think about:

> Defines the parallelism of the Kafka sink operator. By default, the parallelism is determined by the framework using the same parallelism of the upstream chained operator.

File: flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/table/KafkaDynamicSink.java [hunk 2]
```diff
@@ -145,7 +150,17 @@ public SinkRuntimeProvider getSinkRuntimeProvider(Context context) {
 		final FlinkKafkaProducer<RowData> kafkaProducer =
 				createKafkaProducer(keySerialization, valueSerialization);
 
-		return SinkFunctionProvider.of(kafkaProducer);
+		return new SinkFunctionProvider() {
```
Comment (wuchong) [inline flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/table/KafkaDynamicSink.java:153]: Could you add a new util method in `SinkFunctionProvider`? I think most of the connector can use this new method. 

```java
	/**
	 * Helper method for creating a SinkFunction provider with a provided sink parallelism.
	 */
	static SinkFunctionProvider of(SinkFunction<RowData> sinkFunction, Integer sinkParallelism) {
		return new SinkFunctionProvider() {

			@Override
			public SinkFunction<RowData> createSinkFunction() {
				return sinkFunction;
			}

			@Override
			public Optional<Integer> getParallelism() {
				return Optional.ofNullable(sinkParallelism);
			}
		};
	}
```

Comment (zhuxiaoshang) [reply flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/table/KafkaDynamicSink.java:153]: yes,it's better to have common function

**Comment by zhuxiaoshang** [comment]
Thanks @wuchong for your review,I update the pr according to your suggestions.

**Review by wuchong — APPROVED** [review]
LGTM.

**Pull request merged**